src/work-allocation-2/components/case-manager-filter/case-manager-filter.component.ts

```diff
@@ -1,97 +1,151 @@
+          name: 'service',
+          value: ['IA']
+        }
+      ]
     }
-  }
+  };
 
-  @Input()
-  public get jurisdiction(): any {
-    return this.pJurisdiction;
-  }
+  constructor(private readonly filterService: FilterService) {
 
-  public set jurisdiction(value: any) {
-    if (this.pJurisdiction !== value) {
-      this.pJurisdiction = value;
-      this.saveFilterToSession();
-      this.emitChangedEvent();
-    }
   }
 
-  private pLocation: Location = this.ALL_LOCATIONS;
-  private pJurisdiction: any =  {id: 'IA', name: 'Immigration and Asylum'};
-
-  /**
-   * Emit an event to notify the parent component that the selected Caseworker
-   * and/or Location has changed. The new selection is emitted with the event
-   * but can also be retrieved from component.location.
-   */
-  @Output() public selectionChanged: EventEmitter<{ location: Location, jurisdiction: any}>
-    = new EventEmitter<{ location: Location, jurisdiction: any }>();
-
-  /**
-   * Accept the SessionStorageService for adding to and retrieving from sessionStorage.
-   */
-  constructor(private readonly sessionStorageService: SessionStorageService) {
+  private static initServiceFilter(): FilterFieldConfig {
+    return {
+      name: 'service',
+      options: [
+        {
+          key: 'IA',
+          label: 'Immigration and Asylum'
+        }
+      ],
+      minSelected: 1,
+      maxSelected: 1,
+      minSelectedError: 'You must select a service',
+      maxSelectedError: null,
+      title: 'Service',
+      type: 'select'
+    };
   }
 
-  public ngOnChanges(): void {
-    if (this.locations) {
-      // See if we have anything stored in the session for the filter.
-      const stored: string = this.sessionStorageService.getItem(FilterConstants.Session.TaskManager);
-      if (stored) {
-        const { caseworkerId, locationId } = JSON.parse(stored);
-        this.pLocation = this.getLocationById(locationId);
-      }
-      this.emitChangedEvent();
+  private static initCaseLocationFilter(locations: Location[]): FilterFieldConfig {
+    if (!locations) {
+      locations = [];
     }
+    return {
+      name: 'location',
+      options: locations.map(loc => ({key: loc.id, label: loc.locationName})),
+      minSelected: 1,
+      maxSelected: 1,
+      minSelectedError: 'You must select a location',
+      maxSelectedError: null,
+      title: 'Case Location',
+      type: 'select'
+    };
   }
 
-  private saveFilterToSession(): void {
-    const toStore = JSON.stringify({
-      locationId: this.location.id
-    });
-    this.sessionStorageService.setItem(FilterConstants.Session.TaskManager, toStore);
+  private static initRoleTypeFilter(): FilterFieldConfig {
+    return {
+      name: 'role',
+      options: [
+        {
+          key: 'Judicial',
+          label: 'Judicial'
+        },
+        {
+          key: 'Legal Ops',
+          label: 'Legal Ops'
+        },
+        {
+          key: 'Admin',
+          label: 'Admin'
+        }
```

Comment:
It has existing PersonRole which has defined:
export enum PersonRole {
  JUDICIAL = 'Judicial',
  CASEWORKER = 'Legal Ops',
  ADMIN = 'Admin',
  ALL = 'All',
}
Could these be reused?